compiler/ballerina-parser/src/test/resources/expressions/check-on-fail/check_on_fail_source_01.bal

```diff
@@ -0,0 +1,3 @@
+function foo() {
+    check doA() on fail e => error("Something happened", e, c1 = c1, c2 = c2);
```

Comment:
this is actually check-expr within a call-stmt. Let's add direct expression usage here as well. Also, with `checkpanic` keyword.
e.g.
```bal
check doA() on fail e => error("Something happened", e, c1 = c1, c2 = c2);
checkpanic doA() on fail e => error("Something happened", e, c1 = c1, c2 = c2);
_ = check expr on fail err error E("Oops!");
_ = checkpanic expr on fail err error E("Oops!");
```

Reply:
Also add a separate test to test if following can be parsed, as they have ambiguity if we consider only `on-keyword` when parsing `on-fail-check`.

```bal
_ = from int a in b select check c on conflict d;
_ = check from int a in b select check c on conflict d on fail e => error("ERROR!");
_ = from int a in b join C d in e on f equals g select h;
_ = checkpanic from int a in b join C d in e on f equals g select h on fail e => error("Errr!");
```